tests-ui/tests/platform/remote/comfyui/history/adapters/v2ToV1Adapter.test.ts

```diff
@@ -117,4 +121,70 @@ describe('mapHistoryV2toHistory', () => {
       expect(history[0].prompt[3].client_id).toBeUndefined()
     })
   })
+
+  describe('timestamp-based priority assignment', () => {
+    it('assigns priority 0 to items without execution_success timestamp', () => {
+      const result = mapHistoryV2toHistory(historyV2WithMissingTimestamp)
+
+      expect(result).toHaveLength(3)
+
+      const item1000 = result.find(
+        (item) => item.prompt[1] === 'item-timestamp-1000'
+      )
+      const item2000 = result.find(
+        (item) => item.prompt[1] === 'item-timestamp-2000'
+      )
+      const itemNoTimestamp = result.find(
+        (item) => item.prompt[1] === 'item-no-timestamp'
+      )
+
+      expect(item1000).toBeDefined()
+      expect(item2000).toBeDefined()
+      expect(itemNoTimestamp).toBeDefined()
+      if (!item1000 || !item2000 || !itemNoTimestamp) {
+        throw new Error('Expected items not found in result')
+      }
+
+      expect(item2000.prompt[0]).toBe(2)
+      expect(item1000.prompt[0]).toBe(1)
+      expect(itemNoTimestamp.prompt[0]).toBe(0)
+    })
+
+    it('correctly sorts and assigns priorities for multiple items', () => {
+      const result = mapHistoryV2toHistory(historyV2FiveItemsSorting)
+
+      expect(result).toHaveLength(5)
+
+      const item1000 = result.find(
+        (item) => item.prompt[1] === 'item-timestamp-1000'
+      )
```

Comment:
Completely optional: I find that when there's a pattern like this repeated in a test, it's nice to have a name for it, like for this case `findResultByPromptId`

Reply:
I'll submit a follow up.